test/powershell/Modules/Microsoft.PowerShell.Management/Resolve-Path.Tests.ps1

```diff
@@ -48,8 +48,70 @@ Describe "Resolve-Path returns proper path" -Tag "CI" {
             Pop-Location
         }
     }
-    It 'Resolve-Path should support user specified base paths' {
-        $Expected = Join-Path -Path .\ -ChildPath fakeroot
-        Resolve-Path -Path $fakeRoot -RelativeBasePath $testRoot | Should -BeExactly $Expected
+    It 'Resolve-Path RelativeBasePath should handle <Scenario>' -TestCases @(
+        @{
+            Scenario = "Absolute Path, Absolute ReleativeBasePath"
+            Path     = $root
+            Basepath = $testRoot
+            Expected = $root, ".$([System.IO.Path]::DirectorySeparatorChar)fakeroot"
+            CD       = $null
+        }
+        @{
+            Scenario = "Relative Path, Absolute ReleativeBasePath"
+            Path     = ".$([System.IO.Path]::DirectorySeparatorChar)fakeroot"
+            Basepath = $testRoot
+            Expected = $root, ".$([System.IO.Path]::DirectorySeparatorChar)fakeroot"
+            CD       = $null
+        }
+        @{
+            Scenario = "Relative Path, Relative ReleativeBasePath"
+            Path     = ".$([System.IO.Path]::DirectorySeparatorChar)fakeroot"
+            Basepath = ".$([System.IO.Path]::DirectorySeparatorChar)"
+            Expected = $root, ".$([System.IO.Path]::DirectorySeparatorChar)fakeroot"
+            CD       = $testRoot
+        }
+        @{
+            Scenario = "Invalid Path, Absolute ReleativeBasePath"
+            Path     = Join-Path $testRoot ThisPathDoesNotExist
+            Basepath = $root
+            Expected = $null
+            CD       = $null
+        }
+        @{
+            Scenario = "Invalid Path, Invalid ReleativeBasePath"
+            Path     = Join-Path $testRoot ThisPathDoesNotExist
+            Basepath = Join-Path $testRoot ThisPathDoesNotExist
+            Expected = $null
+            CD       = $null
+        }
+    ) -Test {
+        param($Path, $BasePath, $Expected, $CD)
+        
+        if ($null -eq $Expected)
+        {
+            {Resolve-Path -Path $Path -RelativeBasePath $BasePath -ErrorAction Stop} | Should -Throw -ErrorId "PathNotFound,Microsoft.PowerShell.Commands.ResolvePathCommand"
+            {Resolve-Path -Path $Path -RelativeBasePath $BasePath -ErrorAction Stop -Relative} | Should -Throw -ErrorId "PathNotFound,Microsoft.PowerShell.Commands.ResolvePathCommand"
+        }
+        else
+        {
+            try
+            {
+                $OldLocation = if ($null -ne $CD)
+                {
+                    $PWD
+                    Set-Location $CD
+                }
```

Comment:
Uhh :-) Looks like tricky code. Do you mean:
```suggestion
                if ($null -ne $CD)
                {
                    $OldLocation = $PWD
                    Set-Location $CD
                }
```

Reply:
No that was intentional. This ensures `$OldLocation` always gets set so the null check later on works as expected. If I only set it inside the if statement then in scenarios where it evaluates to false, it would inherit the value from the parent scope. This problem can be demonstrated like this:
```
1..10 | . {
    process
    {
        if ($_ % 5 -eq 0)
        {
            $Test = $true
        }

        if ($Test)
        {
            $_
        }
    }
}
```
Now I don't actually know if this applies to Pester but regardless, I consider it a good practice to always make sure a variable gets set before referencing it.